### ISSUE TITLE
Loading compilation database with duplicate files

### ISSUE THREAD

**Issue by Nephyrin** [Andersbakken/rtags#764]
In a project I'm working on, we have a large chunk of files that are built twice, once for a client library and once for a server library.  This causes a few pain points with rtags:
- Only the commandline that appears second is used.  Ideally, rtags would note both configurations for the file, and let one be selected when invoking rc.  This makes things tricky when working with code like:
 ```c++

#ifdef CLIENT_LIBRARY
...
#elif SERVER_LIBRARY
...
#endif
```
... as only one of the two branches will be visible to rtags.
- When re-loading the compilation database, even with no changes, these duplicates files are all invalidated due to being encountered multiple times.

**Comment by Andersbakken** [comment]
Yeah. We have a file or two like this in our codebase at work as well.
Ideally RTags would compile the file twice and combine all the symbols into
one set but it's a bit of a mess. Currently that's not really how it works
but I suppose it could be made to do that.

It's a bit of work butI'll see what I can do.

Anders

**Comment by Nephyrin** [comment]
Is something like this closer to being possible with the multiple-build support? I can generate and load separate compile_commands.json files for the different configurations, but rtags appears to have selected one of the configurations and used it for all diagnostic/completion purposes.

Even with probably-very-difficult merging semantics, it would be very useful to have a configuration-select command to switch between the versions of the file.  Currently this requires switching compilation databases, and triggering much reprocessing.
